gluten-uniffle/velox/src/main/java/org/apache/spark/shuffle/writer/VeloxUniffleColumnarShuffleWriter.java

```diff
@@ -0,0 +1,261 @@
+import org.apache.spark.shuffle.GlutenShuffleUtils;
+import org.apache.spark.shuffle.RssShuffleHandle;
+import org.apache.spark.shuffle.RssShuffleManager;
+import org.apache.spark.shuffle.RssSparkConfig;
+import org.apache.spark.sql.vectorized.ColumnarBatch;
+import org.apache.spark.util.SparkResourceUtil;
+import org.apache.uniffle.client.api.ShuffleWriteClient;
+import org.apache.uniffle.common.ShuffleBlockInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import scala.Option;
+import scala.Product2;
+import scala.collection.Iterator;
+
+public class VeloxUniffleColumnarShuffleWriter<K, V> extends RssShuffleWriter<K, V, V> {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(VeloxUniffleColumnarShuffleWriter.class);
+
+  private long nativeShuffleWriter = -1L;
+
+  private boolean stopping = false;
+  private int compressThreshold = GlutenConfig.getConf().columnarShuffleCompressionThreshold();
+  private double reallocThreshold = GlutenConfig.getConf().columnarShuffleReallocThreshold();
+  private String compressionCodec;
+  private int compressionLevel;
+  private int partitionId;
+
+  private ShuffleWriterJniWrapper jniWrapper = ShuffleWriterJniWrapper.create();
+  private SplitResult splitResult;
+  private int nativeBufferSize = GlutenConfig.getConf().maxBatchSize();
+  private int bufferSize;
+  private PartitionPusher partitionPusher;
+
+  private final ColumnarShuffleDependency<K, V, V> columnarDep;
+  private final SparkConf sparkConf;
+
+  private long availableOffHeapPerTask() {
+    return SparkMemoryUtil.getCurrentAvailableOffHeapMemory()
+        / SparkResourceUtil.getTaskSlots(sparkConf);
+  }
+
+  public VeloxUniffleColumnarShuffleWriter(
+      int partitionId,
+      String appId,
+      int shuffleId,
+      String taskId,
+      long taskAttemptId,
+      ShuffleWriteMetrics shuffleWriteMetrics,
+      RssShuffleManager shuffleManager,
+      SparkConf sparkConf,
+      ShuffleWriteClient shuffleWriteClient,
+      RssShuffleHandle<K, V, V> rssHandle,
+      Function<String, Boolean> taskFailureCallback,
+      TaskContext context) {
+    super(
+        appId,
+        shuffleId,
+        taskId,
+        taskAttemptId,
+        shuffleWriteMetrics,
+        shuffleManager,
+        sparkConf,
+        shuffleWriteClient,
+        rssHandle,
+        taskFailureCallback,
+        context);
+    columnarDep = (ColumnarShuffleDependency<K, V, V>) rssHandle.getDependency();
+    this.partitionId = partitionId;
+    this.sparkConf = sparkConf;
+    bufferSize =
+        (int)
+            sparkConf.getSizeAsBytes(
+                RssSparkConfig.RSS_WRITER_BUFFER_SIZE.key(),
+                RssSparkConfig.RSS_WRITER_BUFFER_SIZE.defaultValue().get());
+    compressionCodec = GlutenShuffleUtils.getCompressionCodec(sparkConf);
```

Comment:
I note celeborn honors Spark's config for SHUFFLE_COMPRESS. Should uniffle do the same check? See https://github.com/apache/incubator-gluten/commit/e182d659b78ef7e384a2020c903dbf65c502012d.

Reply:
Fix